Source/Core/Core/DSP/Jit/x64/DSPJitRegCache.cpp

```diff
@@ -772,6 +772,56 @@ void DSPJitRegCache::PutReg(int reg, bool dirty)
       m_emitter.SAR(64, oparg, Imm8(64 - 40));
     }
     break;
+  case DSP_REG_CR:
+  case DSP_REG_PRODH:
+    if (dirty)
+    {
+      if (oparg.IsSimpleReg())
+      {
+        // register is already shifted correctly
+        // (if at all)
+
+        // Zero extend from the bottom 8 bits.
+        m_emitter.MOVZX(16, 8, oparg.GetSimpleReg(), oparg);
+      }
+      else if (oparg.IsImm())
+      {
+        // TODO: Immediates?
```

Comment:
Perhaps there should be assertions here and in the case below, so if these are ever actually hit they don't silently do nothing

Reply:
Done (also for the previously existing `DSP_REG_ACH0` case).

It's worth noting that `LRI $prod.h,  #0xffff` doesn't hit this case.